splinterd/src/daemon.rs

```diff
@@ -159,40 +160,78 @@ impl SplinterDaemon {
 
         let store_factory = create_store_factory(&self.db_url)?;
 
+        #[cfg(feature = "deprecate-yaml")]
+        {
+            let circuits_location = Path::new(&self.state_dir).join("circuits.yaml");
+            let proposals_location = Path::new(&self.state_dir).join("circuit_proposals.yaml");
+
+            let circuits_location_exists = circuits_location.exists();
+            let proposals_location_exists = circuits_location.exists();
+
+            if circuits_location_exists || proposals_location_exists {
+                error!("splinterd found yaml state files. run the `splinter upgrade` command to import your yaml state into a database");
+
+                if circuits_location_exists {
+                    error!(
+                        "yaml circuits state file: {}",
+                        circuits_location.to_string_lossy()
+                    );
+                }
+
+                if proposals_location_exists {
+                    error!(
+                        "yaml proposals state file: {}",
+                        proposals_location.to_string_lossy()
+                    );
+                }
+
+                panic!("aborting");
```

Comment:
Do not panic here, instead exit with error code 1.

Reply:
Correction, return StarError here.

Reply:
I've changed this to return StartError which propagates up to the main() function and exits with an error code 1. My concern with doing a process exit here was just making sure that the resources were dropped, but I wasn't sure how to weigh that against the error output the user would see. Let me know if I should change it to something else.